x/gamm/pool-models/balancer/amm_test.go

```diff
@@ -638,6 +753,16 @@ func TestCalcJoinPoolShares(t *testing.T) {
 		t.Run(tc.name, func(t *testing.T) {
 			pool := createTestPool(t, tc.swapFee, sdk.MustNewDecFromStr("0"), tc.poolAssets...)
 
+			defer func() {
+				r := recover()
+
+				didPanic := r != nil
+
+				if didPanic != tc.expectPanic {
+					t.Errorf("didPanic: %t, expectPanic: %t", didPanic, tc.expectPanic)
+				}
+			}()
```

Comment:
Can also do suite.Require().Panics or NotPanics, not sure whats easier in concretre instantiation.

Reply:
Yes, please!

Reply:
```go
if `tc.expectPanic` {
 // ...
} else {
  // ...
}
```

Reply:
Done!